server/src/main/java/io/spine/server/route/StateUpdateRouting.java

```diff
@@ -61,10 +53,29 @@ private StateUpdateRouting() {
      *
      * @param <I>
      *         the type of the entity IDs to which the updates are routed
+     * @param idClass
+     *         the class of identifiers served by this routing
      * @return new {@code StateUpdateRouting}
      */
-    public static <I> StateUpdateRouting<I> newInstance() {
-        return new StateUpdateRouting<>();
+    public static <I> StateUpdateRouting<I> newInstance(Class<I> idClass) {
+        return new StateUpdateRouting<>(idClass);
+    }
+
+    private StateUpdateRouting(Class<I> idClass) {
+        super(DefaultStateRoute.newInstance(idClass));
+    }
+
+    /**
+     * Verifies if the passed state type can be routed by a custom route, or
+     * the message has a field matching the type of identifiers served by this routing.
+     */
+    @Override
+    public boolean supports(Class<? extends Message> stateType) {
+        boolean customRouteSet = super.supports(stateType);
+        @SuppressWarnings({"unchecked", "RedundantSuppression"}) // cast to the type used in ctor.
```

Comment:
I don't understand why we suppress a ` "RedundantSuppression"`.

Reply:
Because IDEA does not see this piece as unchecked warning, while the compiler does.